django/db/migrations/state.py

```diff
@@ -403,6 +407,11 @@ def from_model(cls, model, exclude_rels=False):
                 elif name == "index_together":
                     it = model._meta.original_attrs["index_together"]
                     options[name] = set(normalize_together(it))
+                elif name == "indexes":
+                    indexes = model._meta.original_attrs["indexes"]
+                    for index in indexes:
+                        index.name  # Cache the property
```

Comment:
Else the [sanity-check](https://github.com/django/django/pull/6857/files#diff-c8a3e1d248f8f5b10af7ac5b1d8479f4R356) would fail for indexes defined without `name` from `Meta`.